src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -0,0 +1,360 @@
+                    }
+                    catch (final NumberFormatException e)
+                    {
+                        final Set<String> firstNameSet = new HashSet<>();
+                        for (final String citizenName : citizenNameStrings)
+                        {
+                            final String[] citizenNameParts = citizenName.split(" ");
+                            firstNameSet.add(citizenNameParts[0]);
+                        }
+                        final List<String> firstNameMatches = firstNameSet.stream().filter(k -> k.startsWith(potentialArgumentValue)).collect(Collectors.toList());
+                        if (!firstNameMatches.isEmpty())
+                        {
+                            return firstNameMatches;
+                        }
+                    }
+                }
+                else
+                {
+                    if (1 == currentWordIndex)
+                    {
+                        final Set<String> middleNameSet = new HashSet<>();
+                        for (final String citizenName : citizenNameStrings)
+                        {
+                            final String[] citizenNameParts = citizenName.split(" ");
+                            if (potentiaCitizenNameParts[0].equals(citizenNameParts[0]))
+                            {
+                                middleNameSet.add(citizenNameParts[1]);
+                            }
+                        }
+                        return middleNameSet.stream().filter(k -> k.startsWith(potentiaCitizenNameParts[1])).collect(Collectors.toList());
+                    }
+                    else
+                    {
+                        final Set<String> lastNameSet = new HashSet<>();
+                        for (final String citizenName : citizenNameStrings)
+                        {
+                            final String[] citizenNameParts = citizenName.split(" ");
+                            if ((potentiaCitizenNameParts[0].equals(citizenNameParts[0])) && (potentiaCitizenNameParts[1].equals(citizenNameParts[1])))
+                            {
+                                lastNameSet.add(citizenNameParts[2]);
+                            }
+                        }
+                        return lastNameSet.stream().filter(k -> k.startsWith(potentiaCitizenNameParts[2])).collect(Collectors.toList());
+                   }
+                }
+                break;
+        }
+
+        return Collections.emptyList();
+    }
+
+    public Object parse(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @Nullable final BlockPos pos,
+            @NotNull final List<ActionMenuHolder> parsedHolders,
+            final String potentialArgumentValue)
+    {
+        // TODO: selector support, such as used by CommandKill to find player
+        // Entity entity = <net.minecraft.command.CommandBase>.getEntity(server, sender, args[0]);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)